crates/sui-types/src/transaction.rs

```diff
@@ -915,6 +915,14 @@ impl ProgrammableTransaction {
                 value: config.max_programmable_tx_commands().to_string()
             }
         );
+        let total_inputs = self.input_objects()?.len() + self.receiving_objects().len();
```

Comment:
Would you be open to gating this to be bypassed for end of epoch transaction? Might as well happen in one  PR

Reply:
This logic is in `validity_check()` of `ProgrammableTransaction` instead of `TransactionKind`, so it never runs on end of epoch transaction. But is it easy to verify?